core/camel-support/src/main/java/org/apache/camel/support/DefaultComponent.java

```diff
@@ -406,17 +407,25 @@ protected void doInit() throws Exception {
                 name = StringHelper.before(name, ",");
             }
             try {
+                final Registry registry = getCamelContext().getRegistry();
                 log.trace("Discovering optional component property configurer class for component: {}", name);
-                Optional<Class<?>> clazz = getCamelContext().adapt(ExtendedCamelContext.class).getFactoryFinder(RESOURCE_PATH)
-                        .findOptionalClass(name + "-component", null);
-                clazz.ifPresent(c -> componentPropertyConfigurer = org.apache.camel.support.ObjectHelper.newInstance(c, GeneratedPropertyConfigurer.class));
+                componentPropertyConfigurer = registry.lookupByNameAndType(name + "-component", GeneratedPropertyConfigurer.class);
```

Comment:
The name should have -configurer, eg -component-configurer

Reply:
That would introduce a discrepancy between the name of the property file and the name used for the registration of the service in the Camel registry. That would in turn require special code for registering the configurers in Camel Quarkus, which I think we'd like to avoid ( cc @lburgazzoli ?) So if you really want the registry name to have the `-configurer` suffix, could we please also rename the property files?

Reply:
I have added 14c5188 that renames both the property files and the keys used to register the configurers to the registry.

Reply:
The name of the property file is based on its FQN folder name, where its parent folder name is configurer. Not sure what you do on the quarkus side, but you should not change the APT code

Reply:
OK, let me remove the second commit.

Back to your original proposal

> The name should have -configurer, eg -component-configurer

Could you please explain why? I just want to understand your motivation.

Reply:
The registry is end user facing and they can register things too, and some register components with foo-component, etc so you end up with naming clashes too.